std/range/package.d

```diff
@@ -3461,6 +3461,25 @@ pure @safe nothrow unittest
     assert(bd.dropBackOne().equal([1, 2]));
 }
 
+// issue 19823
+@safe unittest
+{
+    import std.format : format;
+    import std.algorithm.iteration : filter;
+
+    auto a = [1, 2, 3, 4];
+    assert(format("%s", a.filter!(a => a != 1).dropOne) == "[3, 4]");
```

Comment:
Maybe drop `format()` and compare `.array`, like done for the last assertion.

Reply:
Hi, Basile-z, Thank you. 

```d
void main() {
    auto a = [1,2,3,4];
    writefln("%s", a.filter!(a => a != 1).dropOne); // [2, 3, 4] WRONG
    writefln("%s", a.filter!(a => a != 2).dropOne); // [3, 4]
    writefln("%s", a.filter!(a => a != 3).dropOne); // [2, 4]
    writefln("%s", a.filter!(a => a != 4).dropOne); // [2, 3]
    writefln("%s", a.filter!(a => a == 1).dropOne); // []
    writefln("%s", a.filter!(a => a == 2).dropOne); // [2] WRONG
    writefln("%s", a.filter!(a => a == 3).dropOne); // [3] WRONG
    writefln("%s", a.filter!(a => a == 4).dropOne); // [4] WRONG
    writefln("%s", a.filter!(a => a != 1).array.dropOne); // [3, 4]
}
```

In this issue, there is no bug in compare `.array`, and the bug will be trigger only when the `dropOne()` operation is performed directly on `filter()` return value `FilterResult`.

Reply:
```d
@safe unittest
{
    import std.algorithm.iteration : filter;
    import std.algorithm.comparison : equal;

    auto a = [1, 2, 3, 4];
    assert(a.filter!(a => a != 1).dropOne.save.equal([3, 4]));
    assert(a.filter!(a => a != 2).dropOne.save.equal([3, 4]));
    assert(a.filter!(a => a != 3).dropOne.save.equal([2, 4]));
    assert(a.filter!(a => a != 4).dropOne.save.equal([2, 3]));
    assert(a.filter!(a => a == 1).dropOne.save.empty);
    assert(a.filter!(a => a == 2).dropOne.save.empty);
    assert(a.filter!(a => a == 3).dropOne.save.empty);
    assert(a.filter!(a => a == 4).dropOne.dropOne.dropOne.save.empty);
    assert(a.filter!(a => a != 1).array.dropOne == [3, 4]);
}
```

Shall I change it to this way? Thank you!

Reply:
sorry, you can mark as resolved. I hadn't see that this was the code from the report.